### ISSUE TITLE
Nextcloud Box version 9.53 Lost Admin password cannot reset

### ISSUE THREAD

**Issue by jmartin17** [nextcloud/nextcloud-snap#120]
When I first power up and got the login screen I put a user ID and password in and logged in successfully. 
When I tried to login on another pc on the LAN it gave me the log in screen but said: Wrong password. Reset it. I tried
again same message so I clicked on the Reset it and it gave me: could not send reset email because there is no email address for this username. I had signed off the PC that I had signed onto when I setup the username and password that I had first logged in successfully.
Now I can’t get back in on any system on the LAN always get same error. I have tried to find out how to reset using browser as there is no keyboard and monitor attached to the NextCloud box and I don't have a way of doing that.
I was able to download PuTTY. 
I can get into the box using PuTTY SSH and I can run commands and see the nextcloud directory but how can I reset or set the password for nextcloud box?
When I use passwd it prompt me for a password but says:
Changing password for ubuntu.

```
(current)
UNIX password:
passwd: Authentication token
manipulation error
passwd: password unchanged
this is trying to change the password for Ubuntu and not nextcloud box admin.
I went ahead and changed the password for the admin and tried again.
I changed the password for ubuntu and it completed successfully. I tried:
login as: ubuntu
ubuntu@192.168.1.183's password:
Welcome to Ubuntu 16.04.1 LTS (GNU/Linux 4.4.0-1017-raspi2 armv7l)
    • Documentation: https://help.ubuntu.com 
    • Management: https://landscape.canonical.com 
    • Support: https://ubuntu.com/advantage 
0 packages can be updated.
0 updates are security updates.
Last login: Mon Aug 29 05:22:22 2016 from 192.168.1.167
ubuntu@ubuntu-standard:~$ sudo nextcloud.occ user:resetpassword jmartin
[sudo] password for ubuntu:
sudo: nextcloud.occ: command not found
the paths to the commands for occ and php are not established.
```

I added path /snap/bin/ to:

```
ubuntu@ubuntu-standard:~$ sudo /snap/bin/nextcloud.occ user:resetpassword jmartin
[sudo] password for ubuntu:
and it returned this error:
An unhandled exception has been thrown:
Doctrine\DBAL\DBALException: Failed to connect to the database: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory in /snap/nextcloud/48/htdocs/lib/private/db/connection.php:54
Stack trace:
#0 /snap/nextcloud/48/htdocs/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(429): OC\DB\Connection->connect()
#1 /snap/nextcloud/48/htdocs/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(389): Doctrine\DBAL\Connection->getDatabasePlatformVersion()
#2 /snap/nextcloud/48/htdocs/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(328): Doctrine\DBAL\Connection->detectDatabasePlatform()
#3 /snap/nextcloud/48/htdocs/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(621): Doctrine\DBAL\Connection->getDatabasePlatform()
#4 /snap/nextcloud/48/htdocs/lib/private/db/connection.php(137): Doctrine\DBAL\Connection->setTransactionIsolation(2)
#5 /snap/nextcloud/48/htdocs/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DriverManager.php(172): OC\DB\Connection->__construct(Array, Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Configuration), Object(Doctrine\Common\EventManager))
#6 /snap/nextcloud/48/htdocs/lib/private/db/connectionfactory.php(116): Doctrine\DBAL\DriverManager::getConnection(Array, Object(Doctrine\DBAL\Configuration), Object(Doctrine\Common\EventManager))
#7 /snap/nextcloud/48/htdocs/lib/private/server.php(368): OC\DB\ConnectionFactory->getConnection('mysql', Array)
#8 /snap/nextcloud/48/htdocs/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#9 /snap/nextcloud/48/htdocs/lib/private/appframework/utility/simplecontainer.php(102): Pimple\Container->offsetGet('DatabaseConnect...')
#10 /snap/nextcloud/48/htdocs/lib/private/servercontainer.php(87): OC\AppFramework\Utility\SimpleContainer->query('DatabaseConnect...')
#11 /snap/nextcloud/48/htdocs/lib/private/server.php(900): OC\ServerContainer->query('DatabaseConnect...')
#12 /snap/nextcloud/48/htdocs/lib/private/server.php(262): OC\Server->getDatabaseConnection()
#13 /snap/nextcloud/48/htdocs/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#14 /snap/nextcloud/48/htdocs/lib/private/appframework/utility/simplecontainer.php(102): Pimple\Container->offsetGet('AppConfig')
#15 /snap/nextcloud/48/htdocs/lib/private/servercontainer.php(87): OC\AppFramework\Utility\SimpleContainer->query('AppConfig')
#16 /snap/nextcloud/48/htdocs/lib/private/server.php(839): OC\ServerContainer->query('AppConfig')
#17 /snap/nextcloud/48/htdocs/lib/private/server.php(413): OC\Server->getAppConfig()
#18 /snap/nextcloud/48/htdocs/3rdparty/pimple/pimple/src/Pimple/Container.php(113): OC\Server->OC\{closure}(Object(OC\Server))
#19 /snap/nextcloud/48/htdocs/lib/private/appframework/utility/simplecontainer.php(102): Pimple\Container->offsetGet('AppManager')
#20 /snap/nextcloud/48/htdocs/lib/private/servercontainer.php(87): OC\AppFramework\Utility\SimpleContainer->query('AppManager')
#21 /snap/nextcloud/48/htdocs/lib/private/server.php(1077): OC\ServerContainer->query('AppManager')
#22 /snap/nextcloud/48/htdocs/lib/private/app.php(254): OC\Server->getAppManager()
#23 /snap/nextcloud/48/htdocs/lib/private/app.php(103): OC_App::getEnabledApps()
#24 /snap/nextcloud/48/htdocs/lib/base.php(665): OC_App::loadApps(Array)
#25 /snap/nextcloud/48/htdocs/lib/base.php(1202): OC::init()
#26 /snap/nextcloud/48/htdocs/console.php(46): require_once('/snap/nextcloud...')
#27 /snap/nextcloud/48/htdocs/occ(11): require_once('/snap/nextcloud...')
#28 {main}ubuntu@ubuntu-standard:~$
```

Any help would be appreciated.

**Comment by kyrofa** [comment]
Hmm, sounds a little fishy. Before we start debugging it though, your box needs to update. It'll do it by itself eventually, but forcing it to do it now might solve your problems.

If you're up for it, run the following commands (while SSHd in):

```bash
$ sudo apt update
$ sudo apt install snapd snap-confine sudo # you can just run apt upgrade here if you prefer
$ sudo snap refresh
```

Once all of these steps have completed, wait for Nextcloud to be visitable again and retry your `occ` reset attempt. Worst case, if you haven't really set it up much, you can always remove and re-install the Nextcloud snap:

```bash
$ sudo snap remove nextcloud && sudo snap install nextcloud
```

Let me know if things start working, or if you need further debugging assistance.

**Comment by jmartin17** [comment]
Kyle:
 
I tried:
 
$ sudo apt update
$ sudo apt install snapd snap-confine sudo # you can just run apt upgrade here if you prefer
$ sudo snap refresh 
 
here is the results:
 
login as: ubuntu
ubuntu@192.168.1.183's password:
Welcome to Ubuntu 16.04.1 LTS (GNU/Linux 4.4.0-1017-raspi2 armv7l)
 
* Documentation:  https://help.ubuntu.com
* Management:     https://landscape.canonical.com
* Support:        https://ubuntu.com/advantage
 
0 packages can be updated.
0 updates are security updates.
 
 
Last login: Fri Sep  2 10:15:41 2016 from 192.168.1.167
ubuntu@ubuntu-standard:~$ sudo apt update
[sudo] password for ubuntu:
Err:1 http://ports.ubuntu.com xenial InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Err:2 http://ports.ubuntu.com xenial-updates InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Err:3 http://ppa.launchpad.net/ubuntu-pi-flavour-makers/ppa/ubuntu xenial InRelease
  Temporary failure resolving 'ppa.launchpad.net'
Err:4 http://ports.ubuntu.com xenial-security InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Err:5 http://ports.ubuntu.com xenial-backports InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Reading package lists... Done
Building dependency tree
Reading state information... Done
All packages are up to date.
W: Failed to fetch http://ports.ubuntu.com/dists/xenial/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ports.ubuntu.com/dists/xenial-updates/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ports.ubuntu.com/dists/xenial-security/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ports.ubuntu.com/dists/xenial-backports/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ppa.launchpad.net/ubuntu-pi-flavour-makers/ppa/ubuntu/dists/xenial/InRelease  Temporary failure resolving 'ppa.launchpad.net'
W: Some index files failed to download. They have been ignored, or old ones used instead.
ubuntu@ubuntu-standard:~$ sudo apt install snapd snap-confine
Reading package lists... Done
Building dependency tree
Reading state information... Done
snap-confine is already the newest version (1.0.38-0ubuntu0.16.04.4).
snap-confine set to manually installed.
snapd is already the newest version (2.12+0.16.04).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
ubuntu@ubuntu-standard:~$ sudo snap refresh
error: cannot list updates: cannot list snaps: cannot list updates: Post https://search.apps.ubuntu.com/api/v1/snaps/metadata: dial tcp: lookup search.apps.ubuntu.com on 208.76.58.193:53: no such host
ubuntu@ubuntu-standard:~$ sudo #
usage: sudo -h | -K | -k | -V
usage: sudo -v [-AknS] [-g group] [-h host] [-p prompt] [-u user]
usage: sudo -l [-AknS] [-g group] [-h host] [-p prompt] [-U user] [-u user]
            [command]
usage: sudo [-AbEHknPS] [-r role] [-t type] [-C num] [-g group] [-h host] [-p
            prompt] [-u user] [VAR=value] [-i|-s] [<command>]
usage: sudo -e [-AknS] [-r role] [-t type] [-C num] [-g group] [-h host] [-p
            prompt] [-u user] file ...
ubuntu@ubuntu-standard:~$ sudo #apt install snapd snap-confine
usage: sudo -h | -K | -k | -V
usage: sudo -v [-AknS] [-g group] [-h host] [-p prompt] [-u user]
usage: sudo -l [-AknS] [-g group] [-h host] [-p prompt] [-U user] [-u user]
            [command]
usage: sudo [-AbEHknPS] [-r role] [-t type] [-C num] [-g group] [-h host] [-p
            prompt] [-u user] [VAR=value] [-i|-s] [<command>]
usage: sudo -e [-AknS] [-r role] [-t type] [-C num] [-g group] [-h host] [-p
            prompt] [-u user] file ...
ubuntu@ubuntu-standard:~$ sudo nextcloud.occ user:resetpassword jmartin
sudo: nextcloud.occ: command not found
ubuntu@ubuntu-standard:~$ sudo /snap/bin/nextcloud.occ user:resetpassword jmartin
The process control (PCNTL) extensions are required in case you want to interrupt long running commands - see http://php.net/manual/en/book.pcntl.php
Enter a new password: bash: /bin/stty: Permission denied
 
bash: /bin/stty: Permission denied
 
Confirm the new password: bash: /bin/stty: Permission denied
 
bash: /bin/stty: Permission denied
                                                                                                                                                             
 
ubuntu@ubuntu-standard:~$ sudo snap remove nextcloud && sudo snap install nextcloud
 
nextcloud removed
error: cannot install "nextcloud": Get https://search.apps.ubuntu.com/api/v1/snaps/details/nextcloud?confinement=strict: dial tcp: lookup search.apps.ubuntu.com on 208.76.58.193:53: no such host
 
ubuntu@ubuntu-standard:~$ sudo snap install nextcloud
error: cannot install "nextcloud": Get https://search.apps.ubuntu.com/api/v1/snaps/details/nextcloud?confinement=strict: dial tcp: lookup search.apps.ubuntu.com on 208.76.58.193:53: no such host
 
I can ping 208.76.58.193 and get a reply
ubuntu@ubuntu-standard:~$ ping 208.76.58.193
PING 208.76.58.193 (208.76.58.193) 56(84) bytes of data.
64 bytes from 208.76.58.193: icmp_seq=1 ttl=53 time=139 ms
64 bytes from 208.76.58.193: icmp_seq=2 ttl=53 time=142 ms
64 bytes from 208.76.58.193: icmp_seq=3 ttl=53 time=138 ms
64 bytes from 208.76.58.193: icmp_seq=4 ttl=53 time=139 ms
64 bytes from 208.76.58.193: icmp_seq=5 ttl=53 time=139 ms
64 bytes from 208.76.58.193: icmp_seq=6 ttl=53 time=140 ms
64 bytes from 208.76.58.193: icmp_seq=7 ttl=53 time=139 ms
64 bytes from 208.76.58.193: icmp_seq=8 ttl=53 time=141 ms
64 bytes from 208.76.58.193: icmp_seq=9 ttl=53 time=144 ms
64 bytes from 208.76.58.193: icmp_seq=10 ttl=53 time=139 ms
64 bytes from 208.76.58.193: icmp_seq=11 ttl=53 time=140 ms
^C
--- 208.76.58.193 ping statistics ---
11 packets transmitted, 11 received, 0% packet loss, time 10014ms
rtt min/avg/max/mdev = 138.869/140.630/144.492/1.652 ms
 
what do I need to correct and how do I do it?
 
James
 
James R. Martin
J. Martin Associates
5185 Tara Ln., Bossier City, LA 71111 - 7810 
Phone: (318) 752-4109 Cell Phone: ( 318) 773-9484 
 
e-mail: jmartin@jmartinassociates.net
 
Web Site:  <http://www.jmartinassociates.net> www.jmartinassociates.net

**Comment by kyrofa** [comment]
@jmartin17 terminal output is brutal to read outside of markdown (which isn't supported in email comments). Please keep that in mind.

It looks like you're having some trouble with DNS. Can you show me the output (nicely formatted, please) of the following commands?

    $ nslookup google.com
    $ dig google.com
    $ nmcli dev show | grep DNS

**Comment by jmartin17** [comment]
`ubuntu@ubuntu-standard:~$ nslookup google.com
-bash: nslookup: command not found
`

`ubuntu@ubuntu-standard:~$ dig google.com
-bash: dig: command not found
`

`ubuntu@ubuntu-standard:~$ nmcli dev show | grep DNS
-bash: nmcli: command not found
`

**Comment by kyrofa** [comment]
`nmcli` was dumb, I forgot you were on a server for a minute. Try installing dnsutils first:

`$ sudo apt install dnsutils`

Then run the `nslookup` and `dig` commands once more.

**Comment by jmartin17** [comment]
I run the `$ sudo apt install dnsutils` and the results was:  

`ubuntu@ubuntu-standard:~$ sudo apt install dnsutils
[sudo] password for ubuntu:
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  bind9-host geoip-database libbind9-140 libdns162 libgeoip1 libisc160
  libisccc140 libisccfg140 liblwres141
Suggested packages:
  rblcheck geoip-bin
The following NEW packages will be installed:
  bind9-host dnsutils geoip-database libbind9-140 libdns162 libgeoip1
  libisc160 libisccc140 libisccfg140 liblwres141
0 upgraded, 10 newly installed, 0 to remove and 0 not upgraded.
Need to get 2939 kB of archives.
After this operation, 9887 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Err:1 http://ports.ubuntu.com xenial/main armhf libgeoip1 armhf 1.6.9-1
  Temporary failure resolving 'ports.ubuntu.com'
Err:2 http://ports.ubuntu.com xenial-updates/main armhf libisc160 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:3 http://ports.ubuntu.com xenial-updates/main armhf libdns162 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:4 http://ports.ubuntu.com xenial-updates/main armhf libisccc140 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:5 http://ports.ubuntu.com xenial-updates/main armhf libisccfg140 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:6 http://ports.ubuntu.com xenial-updates/main armhf libbind9-140 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:7 http://ports.ubuntu.com xenial-updates/main armhf liblwres141 armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:8 http://ports.ubuntu.com xenial-updates/main armhf bind9-host armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:9 http://ports.ubuntu.com xenial-updates/main armhf dnsutils armhf 1:9.10.3.dfsg.P4-8ubuntu1
  Temporary failure resolving 'ports.ubuntu.com'
Err:10 http://ports.ubuntu.com xenial/main armhf geoip-database all 20160408-1
  Temporary failure resolving 'ports.ubuntu.com'
E: Failed to fetch http://ports.ubuntu.com/pool/main/g/geoip/libgeoip1_1.6.9-1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/libisc160_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/libdns162_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/libisccc140_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/libisccfg140_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/libbind9-140_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/liblwres141_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/bind9-host_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/b/bind9/dnsutils_9.10.3.dfsg.P4-8ubuntu1_armhf.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Failed to fetch http://ports.ubuntu.com/pool/main/g/geoip-database/geoip-database_20160408-1_all.deb  Temporary failure resolving 'ports.ubuntu.com'

E: Unable to fetch some archiv`

then I tried to run:  `nslookup google.com` and received `-bash: nslookup: command not found
`
and for `dig google.com` the results was:  `-bash: dig: command not found
`

**Comment by kyrofa** [comment]
Haha, I'm so sorry, of course it was, you're having DNS problems! What is the contents of `/etc/resolv.conf`?

**Comment by jmartin17** [comment]
The content is:

`# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 208.76.58.193
nameserver 208.76.60.152
search Cisco
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
"resolv.conf" [readonly] 5L, 214C                             1,1           All
`

**Comment by kyrofa** [comment]
Try using Google's temporarily:

    $ echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf > /dev/null

Then try to update again. What happens?

    $ sudo apt update

**Comment by jmartin17** [comment]
I ran the `echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf > /dev/null`
it ran OK

The I ran:  `sudo apt update` and it gave me:  `Hit:1 http://ports.ubuntu.com xenial InRelease
Hit:2 http://ppa.launchpad.net/ubuntu-pi-flavour-makers/ppa/ubuntu xenial InRelease
Hit:3 http://ports.ubuntu.com xenial-updates InRelease
Hit:4 http://ports.ubuntu.com xenial-security InRelease
Hit:5 http://ports.ubuntu.com xenial-backports InRelease
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
`

**Comment by kyrofa** [comment]
Excellent, it's probably automatically checking for updates, so that's not a problem. You should be able to see decent results from `$ ping google.com` now, yes?

**Comment by jmartin17** [comment]
yest that worked

**Comment by kyrofa** [comment]
Alright, now run:

    $ sudo apt update
    $ sudo apt upgrade
    $ sudo snap refresh

Your network has some DNS issues. Do you know how to debug those, or do you want to continue using Google's?

**Comment by jmartin17** [comment]
Prolem Resolved NextCloud 10.0.1 installed and running.   Thanks for all your help...
